### ISSUE TITLE
Networking issues

### ISSUE THREAD

**Issue by LEI** [Promaethius/mailcow-dind#11]
First of all I have a limited knowledge about this stuff, so I am here to provide feedback as well as asking for guidance. Also note that I am using `DOCKER_OPTS=--iptables=false --ip-masq=false` on the kubernetes node.

I encountered a networking problem specifically on the watchdog container, basically stopping the build at:

    fetch http://dl-cdn.alpinelinux.org/alpine/v3.8/main/x86_64/APKINDEX.tar.gz

Adding `hostNetwork: true` to the deployment template spec as mentioned in [this alpine issue](https://github.com/gliderlabs/docker-alpine/issues/307) allowed the build to finish.

Probably on the second try at that point, `cmd.sh` started exiting with the message `RTNETLINK answers: File exists` caused by `ip link add dummy0 type dummy` : got around with `ip link delete dummy0` just before.

Those are two minor changes that finally got my pod to run, no idea about side effects.

**Comment by Promaethius** [comment]
@LEI what are you using for a CNI? IPTables is fairly universal to all CNI daemons. This is why adding `hostNetwork: true` would fix the issue. It essentially allows the container to occupy the host network namespace without having to rely on IPTables rules for passing packets in and out of the container namespace.

**Comment by LEI** [comment]
I used Weave Net as my pod network add-on if that's what you are asking.

Previously I had some permission problems with mysql and unbound due to manual volume creation: I suspect the watchdog issue to be caused by the DNS server not being available at this time?

Now that's fixed I got the pod running just fine, and it seems to work without any extra `hostNetwork: true`. I will share my setup if it can be useful to others.
